sql-plugin/src/main/311until320-nondb/scala/org/apache/spark/sql/rapids/shims/RapidsErrorUtils.scala

```diff
@@ -36,10 +36,6 @@ object RapidsErrorUtils {
     new NoSuchElementException(s"Key $key does not exist.")
   }
 
-  def sqlArrayIndexNotStartAtOneError(): ArrayIndexOutOfBoundsException = {
```

Comment:
This seems backwards to me.  IIUC Spark 3.4 decided to fold this exception into another, existing exception.  Since our code base isn't focused on 3.4, it seems odd we're bending all the other Spark's to the 3.4 convention rather than keeping the existing error method and having the 3.4 shim map the `sqlArrayIndexNotStartAtOneError()` to  `QueryExecutionErrors.elementAtByIndexZeroError(context = null)`.  Seems like that would be a lot less change overall and make the calling code more closely match all other Spark versions (except 3.4).

Reply:
>  a lot less change overall and make the calling code more closely match all other Spark versions (except 3.4).

This makes sense, thank you. I will update.

Reply:
Updated.